src/libsyncengine/jobs/network/getappversionjob.cpp

```diff
@@ -0,0 +1,121 @@
+/*
+ * Infomaniak kDrive - Desktop
+ * Copyright (C) 2023-2024 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "getappversionjob.h"
+#include "utility/jsonparserutility.h"
+
+#include <config.h>
+
+namespace KDC {
+GetAppVersionJob::GetAppVersionJob(const Platform platform, const std::string &appID) : _platform(platform), _appId(appID) {
+    _httpMethod = Poco::Net::HTTPRequest::HTTP_GET;
+}
+
+std::string platformToStr(Platform platform) {
+    switch (platform) {
+        case Platform::MacOS:
+            return platformMacOsKey;
+        case Platform::Windows:
+            return platformWindowsKey;
+        case Platform::LinuxAMD:
+            return platformLinuxAmdKey;
+        case Platform::LinuxARM:
+            return platformLinuxArmKey;
+        default:
+            return "unknown";
+    }
+}
+
+std::string GetAppVersionJob::getSpecificUrl() {
+    std::stringstream ss;
+    ss << "/app-information/kstore-update/" << platformToStr(_platform) << "/com.infomaniak.drive/" << _appId;
+    return ss.str();
+}
+std::string GetAppVersionJob::getContentType(bool &canceled) {
+    canceled = false;
+    return {};
+}
+
+bool GetAppVersionJob::handleError(std::istream &is, const Poco::URI &uri) {
+    // TODO
```

Comment:
Is it planned for this PR or for another one?

Reply:
In the next one, sorry. This one was becoming too big due to refactoring of the network job classes.